File: 3295-report-spam-message/3295-report-spam-message.cpp
class Solution {
public:
    bool reportSpam(vector<string>& message, vector<string>& bannedWords) {
        unordered_set<string> banned(bannedWords.begin(), bannedWords.end());
        int count = 0;
        for(const auto& word : message){
            if(banned.find(word) != banned.end()){
                count++;
                if(count >=2) return true;
            }
        }
        return false;
    }
};

File: 3295-report-spam-message/3295-report-spam-message.java
class Solution {
    public boolean reportSpam(String[] message, String[] bannedWords) {
        Set<String> bw=new HashSet<>();
        for (int i=0;i<bannedWords.length;i++){
            bw.add(bannedWords[i]);
        }
        int cou=0;
        for (int i=0;i<message.length;i++){
            if (bw.contains(message[i])){
                cou++;
                if (cou>=2){
                    return true;
                }
            }
        }
        return false;
    }
}